src/gui/org/deidentifier/arx/gui/view/impl/menu/DialogOrderSelection.java

```diff
@@ -193,34 +195,56 @@ private boolean isValidDataType(DataType<?> type, String[] values) {
     }
     
     /**
-     * Loads the array from a file.
+     * Loads the array from a file. If the file contains more or additional
+     * values (lines) than present in the attribute's domain, the loading is
+     * aborted and an <code>IllegalStateException</code> is thrown.
      *
      * @param file
-     * @param charset TODO
+     * @param charset
+     *            TODO
      * @return
+     * @throws IllegalStateException
+     *             The file contains more or additional values (lines) than
+     *             present in the attribute's domain
      */
-    private String[] loadArray(String file, Charset charset) {
-        ArrayList<String> list = new ArrayList<String>();
+    private java.util.List<String> loadFile(String file,

```

Comment:
Please import "java.util.List" and simply use "List" here and in other places where "java.util.List" is explicitly referenced.